gitreload/config.py

```diff
@@ -1,38 +1,45 @@
 """
 Setup configuration from a json file with defaults
 """
-import json
 import logging
 from logging.handlers import SysLogHandler
 import os
 import platform
 
 log = logging.getLogger('gitreload')  # pylint: disable=C0103
 
-CONFIG_PATHS = [
-    os.environ.get('GITRELOAD_CONFIG', ''),
-    os.path.join(os.getcwd(), 'gr.env.json'),
-    os.path.join(os.path.expanduser('~'), '.gr.env.json'),
-    '/etc/gr.env.json',
-]
+MINUTE = 60  # seconds
 
 
-def configure_logging(level_override=None):
+class Config(object):
+    """
+    Configuration for the app
+    """
+    REPODIR = os.environ.get('REPODIR', '/mnt/data/repos')
+    VIRTUAL_ENV = os.environ.get('VIRTUAL_ENV', '/edx/app/edxapp/venvs/edxapp')
+    DJANGO_SETTINGS = os.environ.get('DJANGO_SETTINGS', 'aws')
+    EDX_PLATFORM = os.environ.get('EDX_PLATFORM', '/edx/app/edxapp/edx-platform')
+    LINKED_REPOS = os.environ.get('LINKED_REPOS', {})
+    ALSO_CLONE_REPOS = os.environ.get('ALSO_CLONE_REPOS', {})
+    NUM_THREADS = int(os.environ.get('NUM_THREADS', 1))
+    LOG_LEVEL = os.environ.get('LOG_LEVEL', None)
+    HOSTNAME = platform.node().split('.')[0]
+    LOG_FORMATTER = ('%(asctime)s %(levelname)s %(process)d [%(name)s] '
+                     '%(filename)s:%(lineno)d - '
+                     '{hostname}- %(message)s').format(hostname=HOSTNAME)
+    LOG_FILE_PATH = os.environ.get('LOG_FILE_PATH', '')
+    SUBPROCESS_TIMEOUT = int(os.environ.get('SUBPROCESS_TIMEOUT', 60)) * MINUTE
```

Comment:
Would probably help to call this variable `SUBPROCESS_TIMEOUT_MINUTES` or something. I typically assume that a timeout value is in seconds

Reply:
good point: I'll change that

Reply:
I'll change only the environment variable name and keep the config one as is, given that I translate that in seconds.